apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nav-sidebar/src/store.ts

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/**
+ * External dependencies
+ */
+import { apiFetch, controls, select as triggerSelect } from '@wordpress/data-controls';
+import { combineReducers, registerStore } from '@wordpress/data';
+import { addQueryArgs } from '@wordpress/url';
+import type { Reducer } from 'redux';
+import type { DispatchFromMap, SelectFromMap } from '@automattic/data-stores';
+
+/**
+ * Internal dependencies
+ */
+import { actions, Action } from './actions';
+import { STORE_KEY } from './constants';
+
+export interface Post {
+	id: number;
+	slug: string;
+	status: string;
+	title: string;
+}
+
+const opened: Reducer< boolean, Action > = ( state = false, action ) => {
+	switch ( action.type ) {
+		case 'TOGGLE_SIDEBAR':
+			return ! state;
+
+		default:
+			return state;
+	}
+};
+
+const pages: Reducer< Post[], Action > = ( state = [], action ) => {
+	switch ( action.type ) {
+		case 'RECEIVE_POST_LIST':
+			return action.response.map( ( { id, slug, status, title } ) => ( {
+				id,
+				slug,
+				status,
+				title: title.rendered,
+			} ) );
+
+		default:
+			return state;
+	}
+};
+
+const reducer = combineReducers( { opened, pages } );
+
+type State = ReturnType< typeof reducer >;
+
+const resolvers = {
+	getNavItems: function* () {
+		const currentPostType = yield triggerSelect( 'core/editor', 'getCurrentPostType' );
+		const postType = yield triggerSelect( 'core', 'getPostType', currentPostType );
+
+		const path = `/wp/v2/${ postType.rest_base }`;
+
+		try {
+			const response = yield apiFetch( {
```

Comment:
Is it possible to use `getEntityRecords` here? Just thinking that if the navigation block in Gutenberg [also uses that](https://github.com/wordpress/gutenberg/blob/8885eecb22753c35372c36edb8597315e3a8a431/packages/block-library/src/navigation/edit.js#L307), then we might be able to avoid some of the data being fetched multiple times? That said, that might be an over-optimisation at this early stage, and I see you're already specifying just the fields that you need below, so this query might result in less data / quicker transfer anyway.

Reply:
I'll have to test that out. If `getEntityRecords()` has deep understanding of the `_fields` param, then maybe it'd be able to share the data with the navigation block which'd be good (like maybe `getEntityRecords()` can merge the fields that we both want and send a single request?). But given that I'm requesting this data so early I want the smallest response possible.

Reply:
> But given that I'm requesting this data so early I want the smallest response possible.

Sounds good to me, particularly for sites that might have a lot of pages!

Reply:
Actually that's another reason it's different. We might end up only showing 10 recent posts/pages in the sidebar, and the navigation block might show every page. Maybe sorted by different too. `getEntityRecords()` might be clever enough for that, but maybe not.

Reply:
Good point, that's the tricky thing with abstractions, without reading about its internals, we're not sure if it'll do exactly what you're after here. Sounds like it's safer to stick with `apiFetch` for now.

Reply:
I think I'll still try it out locally at least. It could be that that's the central repository entities in GB, and even if it's not shared by anyone else, that's just the best place to store things.

Reply:
Just switched to `getEntityRecords()` and I was able to delete a bunch of code, so I think this is the way to go 👍

Reply:
Nice, glad it worked out!